### PR TITLE
Set Suite name only if its valid

### LINKED ISSUE #1931: [NPE] Reporter org.testng.reporters.jq.Main failed
### TestNG Version
7.0.0-beta1
Test class contains a single test method with executionCount = 0

### Expected behavior
NPE missing

### Actual behavior
```
Total tests >>>> 5
===============================================
DEFAULT TIGRA API suite
Total tests run: 4, Passes: 4, Failures: 0, Skips: 0
===============================================

[TestNG] Reporter org.testng.reporters.jq.Main@2892d68 failed
java.lang.NullPointerException
	at org.testng.reporters.jq.TimesPanel.maxTime(TimesPanel.java:124)
	at org.testng.reporters.jq.TimesPanel.getContent(TimesPanel.java:89)
	at org.testng.reporters.jq.BaseMultiSuitePanel.generate(BaseMultiSuitePanel.java:25)
	at org.testng.reporters.jq.Main.generateReport(Main.java:74)
	at org.testng.TestNG.generateReports(TestNG.java:1027)
	at org.testng.TestNG.run(TestNG.java:970)
	at org.touchbit.buggy.core.Buggy.run(Buggy.java:138)
	at org.touchbit.buggy.core.Buggy.main(Buggy.java:78)
	at com.ibclabs.aqa.latex.integration.ILatex.main(ILatex.java:9)

```
### Is the issue reproductible on runner?

- [x] Shell
- [ ] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [ ] IntelliJ
- [ ] NetBeans
- [x] Jar

### PR DESCRIPTION
**Pull request by krmahadevan**
Closes #1931

Fixes #1931  .

### Did you remember to?

- [ ] Add test case(s)
- [X] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

**Review by juherr — APPROVED** [review]

**Comment by krmahadevan** [comment]
@cbeust - Can you please help with getting this merged? @juherr has approved the changeset.

File: src/main/java/org/testng/xml/XmlSuite.java [hunk 1]
```diff
@@ -333,7 +334,9 @@ public String getName() {
    * @param name The name to set.
    */
   public void setName(String name) {
-    m_name = name;
+    if (Strings.isNotNullAndNotEmpty(name)) {
+      m_name = name;
+    }
```
Comment (wengm) [inline src/main/java/org/testng/xml/XmlSuite.java]: what else? any warning or exception should be there?

Comment (juherr) [reply src/main/java/org/testng/xml/XmlSuite.java]: Good catch, I agree. An exception could be better than an unexpected behavior.

Comment (krmahadevan) [reply src/main/java/org/testng/xml/XmlSuite.java]: @juherr - added a exception Can we get this merged ?

**Pull request merged**